spec/bank_account_info_spec.rb

```diff
@@ -0,0 +1,118 @@
+      ]
+    end
+
+    subject(:bank_account_info) do
+      described_class.new(
+        account_type: account_type,
+        account_purpose: account_purpose,
+        holder_type: holder_type,
+        holder_tax_id: holder_tax_id,
+        country: country,
+        ispb_code: ispb_code,
+        branch_code: branch_code,
+        account_number: account_number,
+        account_check_digit: account_check_digit,
+        pix_keys: pix_keys
+      )
+    end
+
+    it "requires holder_type, holder_tax_id, branch_code, and account_number" do
+      expect { described_class.new }.to raise_error(ArgumentError)
+
+      expect {
+        described_class.new(
+          holder_tax_id: Incognia::PersonId.new(type: "cpf", value: "12345678901"),
+          branch_code: "0001",
+          account_number: "123456"
+        )
+      }.to raise_error(ArgumentError) # missing holder_type
+
+      expect {
+        described_class.new(
+          holder_type: "business",
+          branch_code: "0001",
+          account_number: "123456"
+        )
+      }.to raise_error(ArgumentError) # missing holder_tax_id
+
+      expect {
+        described_class.new(
+          holder_type: "business",
+          holder_tax_id: Incognia::PersonId.new(type: "cpf", value: "12345678901"),
+          account_number: "123456"
+        )
+      }.to raise_error(ArgumentError) # missing branch_code
+
+      expect {
+        described_class.new(
+          holder_type: "business",
+          holder_tax_id: Incognia::PersonId.new(type: "cpf", value: "12345678901"),
+          branch_code: "0001"
+        )
+      }.to raise_error(ArgumentError) # missing account_number
+
+      expect {
+        described_class.new(
+          holder_type: "business",
+          holder_tax_id: Incognia::PersonId.new(type: "cpf", value: "12345678901"),
+          branch_code: "0001",
+          account_number: "123456"
+        )
+      }.not_to raise_error
+    end
+
+    describe "#to_hash" do
+      it "returns the API format" do
+        expect(bank_account_info.to_hash).to eql(
+          account_type: account_type,
+          account_purpose: account_purpose,
+          holder_type: holder_type,
+          holder_tax_id: {type: "cpf", value: "12345678901"},
+          country: country,
+          ispb_code: ispb_code,
+          branch_code: branch_code,
+          account_number: account_number,
+          account_check_digit: account_check_digit,
+          pix_keys: [{type: "cpf", value: "12345678901"}, {type: "email", value: "human@being.com"}]
+        )
+      end
+    end
+
+    describe "readers" do
+      it "exposes attribute readers" do
+        expect(bank_account_info.account_type).to eq(account_type)
+        expect(bank_account_info.account_purpose).to eq(account_purpose)
+        expect(bank_account_info.holder_type).to eq(holder_type)
+        expect(bank_account_info.holder_tax_id.to_hash).to eq(type: "cpf", value: "12345678901")
```

Comment:
Do we need the `.to_hash` here or we can compare it directly, for example:

```suggestion
        expect(bank_account_info.holder_tax_id).to eq(holder_tax_id)
```

?